File: src/main/services/prompts/context/Memory.ts
import type { PromptModule, PromptContext } from '../PromptTypes'
import { MemoryService } from '../../MemoryService'

export const MemoryModule: PromptModule = {
  id: 'memory',
  layer: 'context',
  priority: 0,
  build: (ctx: PromptContext) => {
    const memDir = MemoryService.getMemoryDir(ctx.workspaceRoot)
    return `# Memory

Persistent memory is stored at \`${memDir}\`. Save only durable user preferences, corrections, project constraints, or external references that will matter in future conversations. Use one focused markdown file per memory and keep a one-line pointer in \`MEMORY.md\`; update existing memories instead of duplicating them.

Do not store facts already recorded by the repository. Treat memory as potentially stale and verify repository state before relying on it.`
  },
}

File: crates/codez-runtime/src/chat/prompt/modules/memory.rs
use crate::chat::prompt::types::{PromptContext, PromptLayer, PromptModule};

pub struct MemoryModule;

use crate::chat::prompt::types::BoxFuture;

impl PromptModule for MemoryModule {
    fn id(&self) -> &'static str {
        "memory"
    }

    fn layer(&self) -> PromptLayer {
        PromptLayer::Context
    }

    fn priority(&self) -> i32 {
        0
    }

    fn is_enabled<'a>(&'a self, _ctx: &'a PromptContext) -> BoxFuture<'a, bool> {
        Box::pin(async { false })
    }

    fn build<'a>(&'a self, _ctx: &'a PromptContext) -> BoxFuture<'a, Option<String>> {
        Box::pin(async { None })
    }
}
